engine/runtime/src/test/java/org/enso/interpreter/test/LazyAtomFieldTest.java

```diff
@@ -122,6 +122,72 @@ public void testInfiniteListGenerator() throws Exception {
     assertEquals(log, 5050, sum.asLong());
   }
 
+  @Test
+  public void fourAtomIntFields() throws Exception {
+    checkNumHolder(
+        """
+    type Num
+        Holder a b c ~num
+
+        new  = Num.Holder 1 2 3 (R.new.nextInt)
+    """);
+  }
+
+  @Test
+  public void fourAtomObjectFields() throws Exception {
+    checkNumHolder(
+        """
+    type Num
+        Holder a b c ~num
+
+        new  = Num.Holder "a" "b" "c" (R.new.nextInt)
+    """);
+  }
+
+  @Test
+  public void fiveAtomIntFields() throws Exception {
```

Comment:
This is the test that simulates #8710 - `fourXYZ` tests were always passing. `fiveXYZ` tests were failing prior to this PR.